transports/upgrades/asbs-4to5.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Azure Service Bus Transport Upgrade Version 4 to 5
+summary: Migration instructions on how to upgrade the Azure Service Bus transport from version 4 to 5
+reviewed: 2025-02-07
+component: ASBS
+related:
+ - transports/azure-service-bus
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 8
+ - 9
+---
+
+Upgrading from Azure Service Bus transport version 4 to version 5 is a major upgrade and requires careful planning. Read the entire upgrade guide before beginning the upgrade process.
+
+Version 5 of the transport introduces the concept of choosing a topic topology. The following two topologies are supported:
+
+- Migration topology
+- Topic per event type topology
+
+The topology selection must be explicitly passed into the constructor of the transport when the transport is being created.
+
+## Topologies
+
+### Topic per Event Type Topology
+
+This topology is the default and preferred choice for new endpoints that do not require backward compatibility with previous versions of the transport. It optimizes event routing, reduces filter overhead, aligns with industry best practices, and improves observability in the event routing path.
+
+The topology represents each event as its own topic. Subscribers add their subscription with forwarding enabled under the topic of the events they are interested in. These subscriptions do not require any filtering rules.
+
+```mermaid
+flowchart TD;
+    Orders -->|Publishes|OrderAccepted;
+    Shipping --> ShippingSubscription -->|Subscribes| OrderAccepted;
+    Shipping -->|ForwardsTo| ShippingQueue;
+    Sales --> SalesSubscription -->|Subscribes| OrderAccepted;
+    Sales -->|ForwardsTo| SalesQueue;
+```
+
+#### Least-privilege
+
+Subscribing and unsubscribing to events require manage rights because entire subscriptions need to be created or deleted. It is possible to run these scenarios with least-privilege access by deploying the necessary subscriptions as part of the endpoint deployment. This can be done by briefly enabling installers, using the [provided tool](/transports/azure-service-bus/operational-scripting.md), or utilizing infrastructure-as-code tools like Bicep, Terraform, or Pulumi.
```

Comment:
```suggestion
Subscribing and unsubscribing to events requires management rights to the Azure Service Bus namespace because subscriptions need to be created or deleted. It is possible to run the transport with least-privilege access by deploying the necessary subscriptions as part of the endpoint deployment. This can be done by briefly enabling installers, using the [provided tool](/transports/azure-service-bus/operational-scripting.md), or utilizing infrastructure-as-code tools such as Bicep, Terraform, or Pulumi.
```